src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.Arrays;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
```

Comment:
It's good that `TruckProducer` is properly parameterized with the `Truck` type, ensuring that no implementation can be created that does not use `Machine` or its subtype.